app/controllers/single_page_subscriptions_controller.rb

```diff
@@ -10,10 +10,13 @@ class SinglePageSubscriptionsController < ApplicationController
 
   skip_before_action :verify_authenticity_token, only: [:show]
 
-  before_action :fetch_subscriber_list, only: %i[show]
+  before_action :fetch_subscriber_list, only: %i[create]
+  before_action :topic_id, only: %i[edit show]
 
-  def show
-    return unless logged_in?
+  def create
+    unless logged_in?
+      redirect_to new_single_page_subscription_path(topic_id: topic_id) and return
```

Comment:
Could this be where the double render error is coming from? I'm not sure what Rails will try to do after it returns from here.

Reply:
Yeah good spot, that wasn't it but this was a line that got lost in my rebase 🤦🏻

Check it out now, you'll see it goes properly to the redirect